### PR TITLE
Artifact registry upload

### PR DESCRIPTION
**Pull request by edman124**
Update BQ pipelines to use BQ 0.4.0 connector
add BQ load gen pipeline

> It's a good idea to open an issue first for discussion.

- [x] Tests pass
- [x] Appropriate changes to documentation are included in the PR

### CODE REVIEW

File: flink-examples-gcp/pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@ under the License.
   <parent>
     <groupId>com.google.flink.connector.gcp</groupId>
     <artifactId>flink-connector-gcp-parent</artifactId>
-    <version>0.0.0</version>
+    <version>0.0.4-SNAPSHOT</version>
```
Comment (clmccart) [inline flink-examples-gcp/pom.xml:27]: this is the version for this package. we probably dont want to update this

File: pom.xml [hunk 2]
```diff
@@ -27,7 +27,7 @@ under the License.
 
     <groupId>com.google.flink.connector.gcp</groupId>
     <artifactId>flink-connector-gcp-parent</artifactId>
-    <version>0.0.0</version>
```
Comment (clmccart) [inline pom.xml:30]: same comment as above

File: flink-examples-gcp/src/main/java/flink/connector/gcp/BQLoadGenerator.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+import com.google.cloud.flink.bigquery.sink.serializer.BigQuerySchemaProvider;
+import com.google.cloud.flink.bigquery.sink.serializer.BigQuerySchemaProviderImpl;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+
+import java.time.Clock;
+import java.util.Random;
+
+/** Pipeline code for running word count reading from Kafka and writing to BQ. */
```
Comment (clmccart) [inline flink-examples-gcp/src/main/java/flink/connector/gcp/BQLoadGenerator.java:46]: update

File: flink-examples-gcp/src/main/java/flink/connector/gcp/BQLoadGenerator.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                Long maxRecords = parameters.getLong("max-records", 1_000_000_000L);
+                String pattern = parameters.get("pattern", "static");
+                Long loadPeriod = parameters.getLong("load-period-in-second", 3600);
+                Long checkpointInterval = parameters.getLong("checkpoint-interval", 60000L);
+                boolean wordGen = parameters.getBoolean("word-gen", true);
+                boolean exactlyOnce = parameters.getBoolean("exactly-once", true);
+                System.out.println("Starting job ".concat(jobName));
+                final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+                env.enableCheckpointing(checkpointInterval);
+                // BQ sink requires checkpointing, which is enabled by default on Big Query Engine for Apache Flink
```
Comment (clmccart) [inline flink-examples-gcp/src/main/java/flink/connector/gcp/BQLoadGenerator.java:69]: let's move this comment up a line

File: flink-examples-gcp/src/main/java/flink/connector/gcp/FailingMapper.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+package flink.connector.gcp;
+
+import org.apache.flink.api.common.functions.MapFunction;
+import org.apache.flink.api.common.state.CheckpointListener;
+
+import org.apache.avro.generic.GenericRecord;
+
+import java.util.Random;
+
+/** Fails 20% of the time after a checkpoint.*/
```
Comment (clmccart) [inline flink-examples-gcp/src/main/java/flink/connector/gcp/FailingMapper.java:28]: can we make the failure rate configurable?

**Review by clmccart — CHANGES_REQUESTED** [review]

File: flink-examples-gcp/src/main/java/flink/connector/gcp/FailingMapper.java (hunk 5 above)
Comment (edman124) [reply flink-examples-gcp/src/main/java/flink/connector/gcp/FailingMapper.java:28]: added a new constructor

**Review by clmccart — APPROVED** [review]

**Pull request merged**